docs/how-to-guides/ale-developer-guides/creating-ale-drivers.md

```diff
@@ -0,0 +1,216 @@
+    ```
+1. Set up an ALE environment
+    - In a terminal, navigate to the base directory of your cloned ALE repo
+    - Create and activate a conda environment:
+        ```sh
+        conda env create ale -f environment.yml
+        conda activate ale
+        ```
+    - Set variables and add ISIS binaries to PATH:
+        ```sh
+        export ISISROOT=[isisRepoLocation]/ISIS3
+        export PATH=$PATH:[isisRepoLocation]/ISIS3/bin
+        ```
+    - Uninstall any preexisting versions of ALE and link in the development version:
+        ```sh
+        pip uninstall ale #run 2 or 3x to ensure that all ALE versions are removed
+        python setup.py develop
+        ```
+
+## Data Setup and Initialization
+
+1. Copy a .cub to a local data area for testing.  See [Data Sources](#data-sources) for a list of data archives.
+1. In your local testing area, create 2 copies of the .cub for testing -- one with ALE and one with ISIS
+        ```sh
+        [cubename].ale.cub
+        [cubename].isis.cub
+        ```
+1. Initialize the ISIS .cub with spiceinit
+        ```sh
+        spiceinit from= [cubename].isis.cub
+        ```
+!!! NOTE "Spiceinit driver order"
+    `spiceinit` uses the first driver that works, first looking for an ALE driver, then using an ISIS driver if all ALE drivers fail.  If you need to initialize a cube with an ISIS driver later, temporarily put a line of code in the ALE driver that will cause it to fail.
+
+## Creating the ALE driver
+### Creating and Naming the Class
+1. If it doesn't exist, create a python file in `ale/base/drivers` for the spacecraft name followed by `_drivers` (e.g. `hayabusa_drivers.py`)
+1. Create a class
+    - See [Class Signature](#class-signature) for naming conventions
+    - Add these common functions (and others as needed):
+        - `instrument_id`
+        - map the sensor name to the spacecraft
+            - spacecraft name as shown in the [NAIF Toolkit Docs](https://naif.jpl.nasa.gov/pub/naif/toolkit_docs/FORTRAN/req/naif_ids.html)
+        - `sensor_model_version`
+            - Should almost always return 11. Add a file for your driver
+!!! TIP "Borrow from existing drivers"
+    ALE drivers largely follow the same format.  If you feel stuck, borrow the structure of another driver and customize it to fit your instrument!
+
+### Creating Methods to Customize the Driver
+ALE drivers largely follow the same structure, but individual methods differ among instruments. The mixins with which your class are built (e.g. "LineScanner") provide a best effort to supply default functionality, but it is unlikely that all of your class's methods will use the default values.  The process of adding methods is largely a matter of trial and error in 1. Getting spiceinit to fully initialize the image using the ALE driver and 2. getting the ALE driver's output to match ISIS's output.
+
+1. Perform an initial best-attempt at filling in the necessary methods (see the common functions above)
```

Comment:
probably best to test with isd_generate.py before testing with ISIS. It should be easier to catch errors early. ISIS testing should be after that.

Reply:
I'm not sure I understand what that workflow looks like.  My understanding is that we have to spiceinit before we can isd_generate.  We don't have anything to isd_generate on until we've already worked through those errors, and at that point we're ready to compare ALE vs ISIS isds.

Reply:
Both ISIS and isd_generate would create the same errors, but ISIS would give the kernels, isd_generate would at best be using meta kernels. I just imagine it would be easier, but on second though, it's not much better.